docs_src/design/adr/core/0019-EdgeX-CLI-V2.md

```diff
@@ -0,0 +1,125 @@
+
+Other issues with the existing client include:
+
+- lack of tab completion
+- default output of commands is too verbose
+- verbose output sometime prevents use of jq 
+- static configuration file required (i.e. no registry support)
+- project hierarchy not conforming to best practice guidelines 
+
+## History
+The original Hanoi V1 client was created by a team at VMWare which is no longer participating in the project. 
+Canonical will lead the development of the Ireland/Jakarta V2 client.
+
+## Decision
+
+1. Use standardized command-line args/flags
+
+| Argument/Flag      | Description |
+| ----------- | ----------- |
+| `-d`, `--debug`      | show additional output for debugging purposes (e.g. REST URL, request JSON, …). This command-line arg will replace -v, --verbose and will no longer trigger output of the response JSON (see -j, --json).       |
+| `-j`, `--json`   | output the raw JSON response returned by the EdgeX REST API and *nothing* else. This output mode is used for script-based usage of the client.    |
+| `--version`   | output the version of the client and if available, the version of EdgeX installed on the system (using the version of the core data service)   |
+
+
+2. Restructure the Go code hierarchy to follow the [most recent recommended guidelines](https://github.com/golang-standards/project-layout). For instance /cmd should just contain the main application for the project, not an implementation for each command - that should be in /internal/cmd
+
+3. Take full advantage of the features of the underlying command-line library, [Cobra](https://github.com/spf13/cobra), such as tab-completion of commands.
+
+4. Change the client binary name from `edgex-cli` to `edgex`. The reason is that the "-cli" is superfluous. There is no `edgex` tool and best practice is to choose a simple, memorable word. CLI tool names do not generally include `"-cli"` in the name. (Examples: `go`,`snap`,`kubectl`,`git`,`docker` etc)
+
+5. Add support for the four common endpoints - `/ping`, `/metrics`, `/version`, `/status`
+
+6. Change the syntax of the CLI to require a positional service arg which precedes the command, thus allowing overlap of command names across services. The service name can either be an abbreviation or the full name of the service . Examples:
+
+    - `edgex data ping`
+    - `edgex d ping`
+    - `edgex metadata version`
+    - `edgex m version`
+    - `edgex metadata device list`
+    - `edgex command device list`
+
+    An alternative solution is to use a positional argument only for the common endpoints but specify it after the command or to have a regular arg like `-s`/`--service` `<name>`).
+
+
+7. Implement all required V2 endpoints for core services
+
+    **Core Command** (`edgex (c)ommand`)
+    - **`command`** `get | put | list` *(we might want to replace get/put with read/write to be more consistent w/other commands)*
+    - **`version`**
+    - **`ping`**
+    - **`metrics`**
+    - **`status`** 
```

Comment:
nit: I would just add a dedicated section for the common endpoints vs. repeating them across all services...